### PR TITLE
tls: making CertificateValidationContextConfigImpl exception free

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: low
Testing: existing tests
Docs Changes: n/a
Release Notes: n/a
Part of https://github.com/envoyproxy/envoy-mobile/issues/176

### CODE REVIEW

File: source/common/ssl/certificate_validation_context_config_impl.h [hunk 1]
```diff
@@ -8,14 +8,26 @@
...
 class CertificateValidationContextConfigImpl : public CertificateValidationContextConfig {
 public:
-  CertificateValidationContextConfigImpl(
-      const envoy::extensions::transport_sockets::tls::v3::CertificateValidationContext& config,
-      Api::Api& api);
+  static absl::StatusOr<std::unique_ptr<CertificateValidationContextConfigImpl>>
+  create(const envoy::extensions::transport_sockets::tls::v3::CertificateValidationContext& context,
+         Api::Api& api) {
+    auto config = std::unique_ptr<CertificateValidationContextConfigImpl>(
+        new CertificateValidationContextConfigImpl(context, api));
```
Comment (ggreenway) [inline source/common/ssl/certificate_validation_context_config_impl.h:22]: make_unique

File: source/common/ssl/certificate_validation_context_config_impl.h [hunk 2]
```diff
@@ -8,14 +8,26 @@
...
 
 class CertificateValidationContextConfigImpl : public CertificateValidationContextConfig {
 public:
-  CertificateValidationContextConfigImpl(
-      const envoy::extensions::transport_sockets::tls::v3::CertificateValidationContext& config,
-      Api::Api& api);
+  static absl::StatusOr<std::unique_ptr<CertificateValidationContextConfigImpl>>
+  create(const envoy::extensions::transport_sockets::tls::v3::CertificateValidationContext& context,
+         Api::Api& api) {
+    auto config = std::unique_ptr<CertificateValidationContextConfigImpl>(
```
Comment (ggreenway) [inline source/common/ssl/certificate_validation_context_config_impl.h:21]: consider moving this to .cc for less work for the compiler

**Review by ggreenway — CHANGES_REQUESTED** [review]
/wait

File: source/common/ssl/certificate_validation_context_config_impl.h (hunk 1 above)
Comment (alyssawilk) [reply source/common/ssl/certificate_validation_context_config_impl.h:22]: you can't with the private constructor.  I prefer the private constructor to make sure folks don't miss the initialize if there are any other callers.

Comment (ggreenway) [reply source/common/ssl/certificate_validation_context_config_impl.h]: Oh right; that's an ugly c++ wart

**Review by ggreenway — APPROVED** [review]

**Pull request merged**